File: Data Structures & Algorithms/rotate-matrix/submission-2.java
class Solution {
    public void rotate(int[][] matrix) {
        int n = matrix.length, m = matrix[0].length;
        for(int i = 0;i<n/2;i++) {
            int[] tmp = matrix[i];
            matrix[i] = matrix[n-1-i];
            matrix[n-1-i] = tmp;
            // swap(matrix[i], matrix[n-1-i]);
        }

        for(int i = 0;i<n;i++) {
            for(int j = i;j<m;j++) {
                int tmp = matrix[i][j];
                matrix[i][j] = matrix[j][i];
                matrix[j][i] = tmp;
            }
        }
    }
}

File: Data Structures & Algorithms/meeting-schedule-ii/submission-2.cpp
/**
 * Definition of Interval:
 * class Interval {
 * public:
 *     int start, end;
 *     Interval(int start, int end) {
 *         this->start = start;
 *         this->end = end;
 *     }
 * }
 */

class Solution {
public:
    int minMeetingRooms(vector<Interval>& intervals) {
        int ans = 0;

        vector<pair<int, int>> tmp;

        for(auto x: intervals) {
            tmp.push_back({x.start, +1});
            tmp.push_back({x.end, -1});
        }

        sort(tmp.begin(), tmp.end(), [](auto& a, auto& b) {
            return a.first == b.first ? a.second < b.second : a.first < b.first;
        });

        int maxi = 0, curr = 0;
        for(auto x: tmp) {
            curr +=x.second;
            maxi = max(maxi, curr);
        }

        return maxi;
    }
};
